tests/pass-dep/libc/libc-random.rs

```diff
@@ -61,3 +63,12 @@ fn test_getrandom() {
         );
     }
 }
+
+#[cfg(any(target_os = "solaris", target_os = "illumos", target_os = "solaris"))]
+fn test_arc4random_buf() {
+    let mut buf = [0u8; 5];
+    unsafe {
+        libc::arc4random_buf(buf.as_mut_ptr(), buf.len());
```

Comment:
`libc` doesn't seem to have this on Solarish targets. For now you can just [copy what `std` does](https://github.com/rust-lang/rust/blob/b9d47cfa9b8f0805bfab2c254a02e598a906f102/library/std/src/sys/random/arc4random.rs#L12-L30).